rpcs3/Emu/Cell/Modules/cellSsl.cpp

```diff
@@ -52,11 +60,15 @@ s32 cellSslCertificateLoader(u64 flag, vm::ptr<char> buffer, u32 size, vm::ptr<u
 	if (required)
 	{
 		*required = (u32)flagBits.count();
-		for (int i = 1; i < flagBits.size(); i++)
+		for (int i = 1; i <= flagBits.size(); i++)
 		{
 			if (!flagBits[i-1])
 				continue;
-			*required += (u32)(getCert(certPath, i).size() - 1);
+			// If we're loading cert 6 (the baltimore cert), then we need set that we're loading the 'normal' set of certs.
+			if (flagBits[5])
+				*required += (u32)(getCert(certPath, i, true).size() - 1);
```

Comment:
Can you please explain to me the reason for those ` - 1` here?

Reply:
Without the `-1`, it reports the wrong size.

Reply:
Hm, and how do you determine that? What is the last char of `getCert` and why it's trimmed here? It's not deleted in the `else` branch. certs are appended as is, but the last cert is skipped for some reason.

Reply:
I figured that out using the tests I linked to below.

Reply:
@hcorion could be a coincidence. Shouldn't `final` here https://github.com/hcorion/rpcs3/blob/4a5454c84602a72a80da43c6a4d7c4586e106fe1/rpcs3/Emu/Cell/Modules/cellSsl.cpp#L86
have always the same `size` (`+ 1` because of `null char`) as a `required`? I.e. `required` should be the size of a buffer able to contain `final` + 1 null char. The operations done in the `if` branch seem like an attempt to much some specific test case.
For example most (if not all) of the `- 1` would be countered by the `flagBits.count()` (hence more or less expected output).

Reply:
> For example most (if not all) of the - 1 would be countered by the flagBits.count()

Yeah, exactly. That's the reason why `required` is not just set to 0 and I add `getCert.size()`. Because the numbers didn't match up with HW/LLE Ssl.

Reply:
@hcorion What is the reason? The numbers should be the same, since you do `- 1` exactly as many times as there are bits in a bitset, so they cancel out (assuming there are no other bugs).

Reply:
No, I subtract `- 1` only when the bit is set to 1. So `required` is equal to size of the string + the number of bits not set.

Reply:
@hcorion cppreference.com/w/cpp/utility/bitset/count `count` return the number of bits that are set. So you just add the number of set bits at first and when subtract it.

Reply:
dang, you're right. Hm, why did I change that... thanks for pointing that out, I've got some testing to do.

Reply:
Done.